cpp/src/structs/utilities.cpp

```diff
@@ -207,73 +213,149 @@ flattened_table flatten_nested_columns(table_view const& input,
   return table_flattener{input, column_order, null_precedence, nullability}();
 }
 
-void superimpose_nulls(bitmask_type const* null_mask,
-                       size_type null_count,
-                       column& input,
-                       rmm::cuda_stream_view stream,
-                       rmm::mr::device_memory_resource* mr)
+namespace {
+
+/**
+ * @brief Dispatcher to conservatively check if the input column needs to be sanitized (i.e., if it
+ * may contain non-empty nulls).
+ *
+ * For performance reason, no non-empty null will actually be checked. Instead, this will perform
+ * conservative checking to see if the input column contains any lists or strings column that is
+ * nulllable. As such, it may return false positive answer.
+ */
+struct may_need_sanitize_dispatch {
+  template <typename T>
+  bool operator()(column_view const& col) const
+  {
+    return std::any_of(col.child_begin(), col.child_end(), [](auto const& child) {
+      return type_dispatcher(child.type(), may_need_sanitize_dispatch{}, child);
+    });
+  }
+};
```

Comment:
I think this might need rephrasing, because I'm not sure we have defined what "sanitize" means in this context. I agree that you and I know it, but it might not be clear to the lay reader.
"Purge" is a clearer term (as you've used in the PR's title), but "sanitize" is more handy. If we defined "sanitize == purge nested contents for null rows" somewhere, that would be useful.

Reply:
Thanks. I'll rename it to `may_contain_non_empty_nulls` which is clearer.